File: backend/src/main/java/com/freshtrack/backend/service/AuthService.java
package com.freshtrack.backend.service;

import com.freshtrack.backend.dto.auth.AuthResponse;
import com.freshtrack.backend.dto.auth.LoginRequest;
import com.freshtrack.backend.dto.auth.RegisterRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: frontend/src/services/authService.ts
import type { AuthResponse, LoginRequest, RegisterRequest } from "../types/auth.ts";
import api from "../api/axios.ts";

export const login = async (
    request: LoginRequest
): Promise<AuthResponse> => {

    const response = await api.post<AuthResponse>(
        "/auth/login",
        request
    );

    return response.data;
};

export const register = async (
    request: RegisterRequest
): Promise<AuthResponse> => {

    const response = await api.post<AuthResponse>(
        "/auth/register",
        request
    );

    return response.data;
};